File: 22/day1.py
def openFile(filename):
    '''Takes in a string containing a file name,
    returns an array of the lines of the file.'''
    f = open(filename)
    lines = f.readlines()
    f.close()

    return lines

def solution(lines, part2):
    '''Takes in a string containing the lines of the file representing the
    calories and whether or not it is part2, returns the highest or top 3
    highest calories.'''
    current = 0
    top = [0, 0, 0]

    for i in range(len(lines)):
        lines[i] = lines[i].replace('\n', '')

        # Still in the same group of ints
        if lines[i]:
            current += int(lines[i])
        # Empty line representing end of the group
        else:
            if top[0] < current:
                top[2], top[1], top[0] = top[1], top[0], current
            elif top[1] < current:
                top[2], top[1] = top[1], current
            elif top[2] < current:
                top[2] = current
            current = 0

    if not part2:
        return  top[0]
    else:
        return (top[0] + top[1] + top[2])

f = openFile('day1.txt')

print("Solution 1:", solution(f, False))
print("Solution 2:", solution(f, True))

File: 22/day1.c
#include <stdio.h>

void main()
{
    FILE* fp = fopen("day1.txt", "r");

    int current_calories = 0;
    int top[] = {0, 0, 0};
    char line[10];
    
    while(fgets(line, 10, fp))
    {
        if( line[0] == '\n' || line[0] == '\r' )
        {
            if( top[0] < current_calories )
            {
                top[2] = top[1];
                top[1] = top[0];
                top[0] = current_calories;
            }
            else if( top[1] < current_calories )
            {
                top[2] = top[1];
                top[1] = current_calories;
            }
            else if( top[2] < current_calories )
            {
                top[2] = current_calories;
            }
            current_calories = 0;
        }
        else
        {
            current_calories += atoi(line);
        }
    }

    printf("Part 1 Solution: %d\n", top[0]);
    printf("Part 2 Solution: %d\n", top[0] + top[1] + top[2]);

    fclose(fp);

}